internals/secrethub/credential_backup_test.go

```diff
@@ -0,0 +1,128 @@
+package secrethub
+
+import (
+	"bytes"
+	"errors"
+
+	"github.com/secrethub/secrethub-go/internals/assert"
+
+	"strings"
+
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/credentials"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/fakeclient"
+
+	"testing"
+
+	"github.com/secrethub/secrethub-cli/internals/cli/ui/fakeui"
+	"github.com/secrethub/secrethub-go/pkg/secrethub"
+)
+
+func TestCredentialBackupCommand_Run(t *testing.T) {
+	var testError = errors.New("test")
+
+	testCases := map[string]struct {
+		cmd            CredentialBackupCommand
+		promptOut      string
+		out            string
+		in             string
+		username       string
+		createError    error
+		meError        error
+		newClientError error
+		promptError    error
+		err            error
+	}{
+		"fail-client-error": {
+			err:            testError,
+			newClientError: testError,
+		},
+		"fail-me-error": {
+			err:     testError,
+			meError: testError,
+		},
+		"fail-abort": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in:  "n",
+			out: "Aborting\n",
+		},
+		"fail-create-error": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in:          "y",
+			err:         testError,
+			createError: testError,
+		},
+		"fail-no-account-key": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
+			in:  "y",
+			err: errors.New("backup code has not yet been generated"),
+		},
+
+		"success": {
+			cmd: CredentialBackupCommand{},
+			promptOut: "This will create a new backup code for Chucky. " +
+				"This code can be used to obtain full access to your account.\n" +
+				"Do you want to continue? [Y/n]: ",
```

Comment:
We could avoid repeating this string by declaring it as a constant for the test.